_posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc

```diff
@@ -0,0 +1,1927 @@
+that the code does what we say it does—and that it continues to do it over years of updates.
+
+=== Versioning
+
+All of the code in the Gruntwork Service Catalog is _versioned_. Every time we make a change, we put out a new
+versioned release, and announce it in the monthly
+https://blog.gruntwork.io/tagged/gruntwork-newsletter[Gruntwork Newsletter].
+
+When you use the code from the Gruntwork Service Catalog (a topic we'll cover in <<how_to_use_the_catalog>>), you pin
+yourself to a specific version of the code. That way, you are not accidentally affected by any subsequent changes in
+the Gruntwork Service Catalog until you explicitly choose to pull those changes in. And when you do want to pull the
+changes in, it's just a matter of bumping the version number!
+
+We use version numbers of the form `MAJOR.MINOR.PATCH` (e.g., `1.2.3`), following the principles of
+_https://semver.org[semantic versioning]_. In traditional semantic versioning, you increment the:
+
+. MAJOR version when you make incompatible API changes,
+. MINOR version when you add functionality in a backwards compatible manner, and
+. PATCH version when you make backwards compatible bug fixes.
+
+However, much of the Gruntwork Service Catalog is built on Terraform, and as Terraform is still not at version `1.0.0`
+(latest version as of August, 2019, was `0.12.6`), most of the Gruntwork Service Catalog is using `0.MINOR.PATCH`
+version numbers. With `0.MINOR.PATCH`, the rules are a bit different, where you increment the:
+
+. MINOR version when you make incompatible API changes
+. PATCH version when you add backwards compatible functionality or bug fixes.
+
+=== The Gruntwork Reference Architecture
+
+// TODO: Ref Arch diagram
+
+The https://gruntwork.io/reference-architecture/[Gruntwork Reference Architecture] is a production-grade, end-to-end
+tech stack built on top of the modules from the Gruntwork Service Catalog.
+
+What's included::
+  The Reference Architecture includes just about everything the typical company needs: multiple environments, each
+  configured with server orchestration (e.g., Kubernetes), load balancers, databases, caches, network topology,
+  monitoring, alerting, log aggregation, CI/CD, user management, secrets management, SSH management, VPN management, and
+  much more (for a detailed walkthrough, see
+  https://blog.gruntwork.io/how-to-build-an-end-to-end-production-grade-architecture-on-aws-part-1-eae8eeb41fec[How to Build an End to End Production-Grade Architecture on AWS]).
+  We wire all these pieces together according to your needs, deploy it into your AWS or GCP accounts, and give you
+  100% of the code—all in about one day.
+
+Opinionated code::
+  Whereas the Gruntwork Service Catalog is relatively unopinionated, allowing you to combine and compose modules, tools,
+  and approaches however you want ("a la carte"), the Gruntwork Reference Architecture is more opinionated, giving you a
```

Comment:
```suggestion
  and approaches however you want ("à la carte"), the Gruntwork Reference Architecture is more opinionated, giving you a
```

Reply:
I KNEW IT WOULD BE USEFUL ONE DAY
![](https://media.giphy.com/media/KKI3fqPszOl3i/giphy.gif)

Reply:
Flight of the Conchords reference ❤️

Reply:
Thx for the fix! And doubly sow for the excellent `.gif` 😁

Reply:
We played foux du fafa at least once a week at my previous office for some reason :joy: